ocean_lib/ocean/ocean.py

```diff
@@ -335,33 +333,37 @@ def create_pool(
         return bpool
 
     @enforce_types
-    def build_compute_provider_fees(
+    def retrieve_provider_fees(
         self,
-        provider_data: Union[str, dict],
-        provider_fee_address: str,
-        provider_fee_token: str,
-        provider_fee_amount: int,
-        valid_until: int,
+        metadata: dict,
+        publisher_wallet: Wallet,
+        encrypted_files: Any,
+        erc721_nft: ERC721NFT,
+        erc20_token: ERC20Token,
     ) -> tuple:
-        if isinstance(provider_data, dict):
-            provider_data = json.dumps(provider_data, separators=(",", ":"))
-
-        message = self.web3.solidityKeccak(
-            ["bytes", "address", "address", "uint256", "uint256"],
-            [
-                self.web3.toHex(self.web3.toBytes(text=provider_data)),
-                provider_fee_address,
-                provider_fee_token,
-                provider_fee_amount,
-                valid_until,
-            ],
+        data_provider = DataServiceProvider
+        ddo = self.assets.create(
```

Comment:
The `retrieve_provider_fees` method should probably not call the `assets.create()` method. 

Publishing the asset happens once.  But you'll need to retrieve provider fees potentially many times.